### ISSUE TITLE
[bug] Adepta Sororitas.cat - Incorrect Model/Upgrade tags across the whole army 

### ISSUE THREAD

**Issue by Nokhal** [BSData/wh40k-7th-edition#1364]
Overall problem : Incorrect model count/upgrade total with battlescribe pro; 

and fucked up Retributor squad creation + missing check on the command one.

Bug/root + quicfix given here.

----------------------------------- Battle sisters squad -------------------------------------------
The sister Superior type should be model, not upgrade.
TODO : 
-Set battle sister squad price to 0
-Set Type of Sister superior to model
-Set price of Sister superior to 12
-Remove the direct profile of the sister superior. 
-Create a hide condition under the link of the profile of the sister superior if she has the option "upgraded to veteran"

![like_this](https://cloud.githubusercontent.com/assets/13810287/9289625/aca72904-4377-11e5-9489-be63d48844b4.png)

----------------------------------- Celestian squad -------------------------------------------
The Celestian Superior type should be model, not upgrade.
TODO : 
-Set Celestian squad price to 0
-Set Type of Sister superior to model
-Set price of Celestian superior to 14

----------------------------------- Dominion squad -------------------------------------------

The sister Superior type should be model, not upgrade.
TODO : 
-Set Dominion squadsquad price to 0
-Set Type of Sister superior to model
-Set price of Sister superior to 13
-Create a hide condition (add modifier, hide, then add condition) under the link of the profile of the Dominion superior if she has the option "upgraded to veteran"

----------------------------------- Retributor Squad -------------------------------------------
Very ambiguous in what you are taking, displayed price is options+model (instead of just options), completely fucks up the upgrade/model ratios. Solution is to look at how the battle sister squad was made.
TODO : 
-Create a new entry called "Retributor", type: model, 12 points, Min Selections 4, Max slection 9.
-Move the Heavy Weapons entry to the "Retributor squad" entry.
-Set all heavy weapons price to 10.
-Copy the boltgun entry from the battle sister squad to the "Retributor squad".
-Set the Retributor superior entry type to "Model"
-Create the Retributor profile hiding condition in the same way the basic squad and sister of dominions are hidden
-Move botlgun from "Retributor with" to "Retributor squad"
-Delete the hold "Retributor with" entry.
-Makes sure you still have the simulacrum imperialis entry link in the squad profile

----------------------------------- Seraphim Squad -------------------------------------------
The sister Superior type should be upgrade, not model.
TODO : 
-Create a "May upgrade one Seraphim to a Seraphim Superior" entry group
-Move the old seraphim superior entrance under it (cut/paste)
-Change the seraphim superior type to upgrade

Incorrect number of bolt pistol displayed
TODO : 
Move the "bolt pistol" entry from under the "Seraphim" entry to under the "Seraphim Squad" entry. untick "Collective".

----------------------------------- Sororitas Command squad -------------------------------------------
This one is tricky.
It's faster to start again from scratch.
I'll open a separate issue, as at least the entry we have right now let you select the correct numbers of points (but in a totally counter intutive way involving deleting models in the interface).

**Comment by Nokhal** [comment]
Also i already fixed them in my local cat but i'm too lazy to do all the github setup.
I can pastebin it to you if you want.

**Comment by amis92** [comment]
You can try to anonymously send us this file using appspot site (http://battlescribedata.appspot.com/#/repo/wh40k)

**Comment by Nokhal** [comment]
There was an error submitting your file. Please try again later.

**Comment by Nokhal** [comment]
https://github.com/BSData/wh40k/issues/1364 fixed + HQ command squad updated.

-Correct number of model displayed for the army using battlescribe pro.
-Redone Command Squad and retributor squad to match the codex wording and process of selection (no more 22 points ambiguous bolter choice for retributor, nor more manually adjusting the number of celestian in a squad that is supposed to always have 5 members.
-Fixed profile hiding of retributor superior. 
-Fixed number of bolt pistol in seraphim squad.
-Fixed the number of selectable banners in a command squad. Still exploitable by creating 2 command squad and putting two banner in one. See : https://github.com/BSData/wh40k/issues/1365
Will be transparent for most player. Allows the creation of illegal armies, does not mark legal armies as illegal. 

Dl of my V30 of the dex, I tested it with various composition and battle formation.
http://dl.free.fr/itSBNPLP8

**Comment by amis92** [comment]
We try to keep consistency in transferring Codex entries in such a way that sergeant/justicar/leader/superiors are models and not upgrades. Correct naming thereof is unfortunately not always done.

**Comment by Nokhal** [comment]
Well, for sister, the Superior is a model, included in the base price of the unit -not optionnal, same profile as the rest of the squad except that it's a character+gear options-, the veteran is an upgrade.

**Comment by amis92** [comment]
Unless the model is the unit as is for tanks (but not tank squadrons), the unit should not have any "included" cost. The unit consists of models, and each one of them costs a specific amount of points. Superior also has it's cost. The wording they use in modern codexes doesn't change it.

**Comment by Nokhal** [comment]
Then I fixed it the correct way. No unit in my update of the codex has a cost, everything is either in the models or their upgrades. Except the repentia squad, because a model cost is unclear.

**Comment by amis92** [comment]
Great then. Hard to read raw XML, didn't understand correctly what you did. ;)

**Comment by Nokhal** [comment]
Basically lot's of clean up following the guideline in my first post + redone retributors and HQ squad. 

Updated file with 0 point unit for repentia -points transferred to the repentia superior- and type "model" for the exorcist. 

http://dl.free.fr/rFl0eB0rf

**Comment by hisop** [comment]
Closing this thread if contributions are complete.

**Comment by amis92** [comment]
@hisop I think it was never actually merged.

**Comment by amis92** [comment]
@hisop As you've taken tidy up on yourself per #1674 , and as far as I see this contribution wasn't merged: do what you want to with it, but I think it might be useful. :)

**Comment by hisop** [comment]
@amis92 Thanks for this, i noticed also, some of the changes suggested aren't quite accurate but most are valid (I mostly closed this so i could finish off the changes since it has been open several months) i will finish all three together then make a separate task for referencing, then tidy up any other sisters bugs or requests.

**Comment by amis92** [comment]
Okay, great :) Well then, I'll leave you to it.

**Comment by hisop** [comment]
@amis92 I have "fixed" the command squad but he raises a good point  " totally counter intutive way involving deleting models in the interface" now the fixed version as bellow has the correct minimum value displayed as celestians are default, but you have to "remove" a celestian from the interface to add any of the upgrade characters, is it better to preserve minimum cost, or leave default entry blank/actual minimum?
![image](https://cloud.githubusercontent.com/assets/7855347/11815799/612cb8f8-a345-11e5-9027-43907135455c.png)

**Comment by amis92** [comment]
Okay, now I may be doing this over-the-top, but if I understand correctly, you must have 5 models is squad, and each upgrade character can be taken only once.

So what if you made 5 separate Entry Groups, one for each model, and each group would allow for choice of Celestian or upgrade (different one in each group). Ofc the default would be Celestian.

That way you have both correct initial cost, no 'model removing', and all options allowed. I seems to be a bit overkill, but I think it's a good idea. Groups could be named as 'Upgrade to Dialogus', for example.

**Comment by hisop** [comment]
I see what your are saying it is a good idea, i have a good idea for a solution based on this principle  now thank you.
